src/ML-METATENSOR/pair_metatensor.cpp

```diff
@@ -0,0 +1,511 @@
+    if (argc == 0) {
+        error->all(FLERR, "expected at least 1 argument to pair_style metatensor, got {}", argc);
+    }
+
+    const char* model_path = argv[0];
+    const char* extensions_directory = nullptr;
+    const char* requested_device = nullptr;
+    for (int i=1; i<argc; i++) {
+        if (strcmp(argv[i], "check_consistency") == 0) {
+            if (i == argc - 1) {
+                error->all(FLERR, "expected <on/off> after 'check_consistency' in pair_style metatensor, got nothing");
+            } else if (strcmp(argv[i + 1], "on") == 0) {
+                this->check_consistency = true;
+            } else if (strcmp(argv[i + 1], "off") == 0) {
+                this->check_consistency = false;
+            } else {
+                error->all(FLERR, "expected <on/off> after 'check_consistency' in pair_style metatensor, got '{}'", argv[i + 1]);
+            }
+
+            i += 1;
+        } else if (strcmp(argv[i], "extensions") == 0) {
+            if (i == argc - 1) {
+                error->all(FLERR, "expected <path> after 'extensions' in pair_style metatensor, got nothing");
+            }
+            extensions_directory = argv[i + 1];
+            i += 1;
+        } else if (strcmp(argv[i], "device") == 0) {
+            if (i == argc - 1) {
+                error->all(FLERR, "expected string after 'device' in pair_style metatensor, got nothing");
+            }
+            requested_device = argv[i + 1];
+            i += 1;
+        } else {
+            error->all(FLERR, "unexpected argument to pair_style metatensor: '{}'", argv[i]);
+        }
+    }
+
+    this->load_torch_model(model_path, extensions_directory);
+
+    // Select the device to use based on the model's preference, the user choice
+    // and what's available.
+    auto available_devices = std::vector<torch::Device>();
+    for (const auto& device: this->capabilities->supported_devices) {
+        if (device == "cpu") {
+            available_devices.push_back(torch::kCPU);
+        } else if (device == "cuda") {
+            if (torch::cuda::is_available()) {
+                // Get a MPI communicator for all processes on the current node
+                MPI_Comm local;
+                MPI_Comm_split_type(world, MPI_COMM_TYPE_SHARED, 0, MPI_INFO_NULL, &local);
+                // Get the rank of this MPI process on the current node
+                int local_rank;
+                MPI_Comm_rank(local, &local_rank);
+
+                int size;
+                MPI_Comm_size(local, &size);
+                if (size < torch::cuda::device_count()) {
+                    if (comm->me == 0) {
+                        error->warning(FLERR,
+                            "found {} CUDA-capable GPUs, but only {} MPI processes on the current node; the remaining GPUs will not be used",
+                            torch::cuda::device_count(), size
+                        );
+                    }
+                }
```

Comment:
Is this safe when some devices are hidden? Some people set one visible device per process

Reply:
We might also tell people to avoid using `VISIBLE_DEVICES`

Reply:
Yeah, I did not consider that. I'll check if we can know whether the user explicitly set `CUDA_VISIBLE_DEVICE` or something else, and send the warning only if they did not.

Reply:
In general, I need to check this code a bit more in presence of `CUDA_VISIBLE_DEVICE`

Reply:
Ok, so after thinking more carefully, this code should be fine even in presence of `CUDA_VISIBLE_DEVICE`. 

If it is set, `torch::cuda::device_count()` will be 1, so this warning will never trigger (since `size` is at least 1); and `gpu_to_use` below will be set to 0, using the first device, which will be the one from `CUDA_VISIBLE_DEVICE`.